### PR TITLE
feat(contracts): handles revert better

### PR DESCRIPTION
**Pull request by jurajpiar**
Resolves https://rsklabs.atlassian.net/browse/RMKT-887
Screenshot 2021-08-11 at 11 05 49

### CODE REVIEW

File: src/contracts/utils/waitForReceipt.ts [hunk 1]
```diff
@@ -14,11 +14,17 @@ function waitForReceipt(
...
 
-      if (receipt != null) {
+      if (receipt?.status) {
         clearInterval(checkInterval)
         resolve(receipt)
       }
 
+      if (receipt && !receipt.status) {
+        reject(
+          new Error(`Transaction reverted, ${web3.eth.handleRevert && 'please, see the "MetaMask - RPC Error" bellow, under "data.error.message" for details.'}  receipt: ${JSON.stringify(receipt, null, 2)}`),
```
Comment (itofarina) [inline src/contracts/utils/waitForReceipt.ts:24]:
```suggestion
          new Error(`Transaction failed, ${web3.eth.handleRevert && 'please, see the "RPC Error" bellow, under "data.error.message" for details.'}  receipt: ${JSON.stringify(receipt, null, 2)}`),
```

**Review by itofarina — APPROVED** [review]
Good catch!

**Review by vjgene — APPROVED** [review]
Nice fix!

File: src/contracts/utils/waitForReceipt.ts (hunk 1 above)
Comment (jurajpiar) [reply src/contracts/utils/waitForReceipt.ts:24]: The `receipt.status` is only false when reverted. https://web3js.readthedocs.io/en/v1.2.7/web3-eth.html?highlight=status#eth-gettransactionreceipt-return

**Pull request merged**